src/rest.cpp

```diff
@@ -195,15 +193,27 @@ static bool rest_headers(const std::any& context,
     std::vector<std::string> path;
     boost::split(path, param, boost::is_any_of("/"));
 
-    if (path.size() != 2)
-        return RESTERR(req, HTTP_BAD_REQUEST, "No header count specified. Use /rest/headers/<count>/<hash>.<ext>.");
+    std::string raw_count {};
+    std::string hashStr {};
+    if (path.size() == 2) {
+        // deprecated path: /rest/headers/<count>/<hash>
+        hashStr = path[1];
+        raw_count = path[0];
+    }
+    else if (path.size() == 1) {
+        // new path with query parameter: /rest/headers/<hash>?count=<count>
+        hashStr = path[0];
+        raw_count = req->GetQueryParameter("count", "5");;
+    }
+    else {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid request. Use /rest/headers/<hash>.<ext>?count=<count>");
+    }
 
-    const auto parsed_count{ToIntegral<size_t>(path[0])};
+    auto parsed_count = ToIntegral<size_t>(raw_count);
     if (!parsed_count.has_value() || *parsed_count < 1 || *parsed_count > MAX_REST_HEADERS_RESULTS) {
-        return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Header count is invalid or out of acceptable range (1-%u): %s", MAX_REST_HEADERS_RESULTS, path[0]));
+        return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Header count is invalid or out of acceptable range (1-%u): %s", MAX_REST_HEADERS_RESULTS, raw_count));
     }
 
-    std::string hashStr = path[1];
     uint256 hash;
     if (!ParseHashStr(hashStr, hash))
         return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + hashStr);
```

Comment:
if I try to pass a blockhash without the extension (.<bin|hex|json>) in `/headers`, like:
`/rest/headers/000000001aeae195809d120b5d66a39c83eb48792e068f8ea1fea19d84a4278a?count=5`
returns: `Invalid hash: 000000001aeae195809d120b5d66a39c83eb48792e068f8ea1fea19d84a4278a?count=5`

In the deprecated, `/rest/headers/5/000000001aeae195809d120b5d66a39c83eb48792e068f8ea1fea19d84a4278a`
returns: `output format not found (available: .bin, .hex, .json)`

Reply:
You're right, that should be handled better. I've:
- updated `ParseDataFormat` to ignore the query string even when no data format is found
- updated documentation on `ParseDataFormat` in `rest.h` to document existing and new behaviour
- added a test case to `test_query_string` to capture what you've found

Rebased and pushed to 0415aaafb

Reply:
Tested! Works!